provider/kubernetes/percentage.go

```diff
@@ -0,0 +1,44 @@
+package kubernetes
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// PercentageValue is int64 form of percentage value with 10^-3 precision.
+type PercentageValue int64
+
+// PercentageValueFromString tries to read percentage value from string like "1.1%", "6%".
+// It will lose the extra precision if there are more than 3 digit after decimal point.
+func PercentageValueFromString(s string) (PercentageValue, error) {
```

Comment:
Could you not use a variadic function for precision, so that it could be configured to allow for more digits if required?  `(s string, precision ...int)` ?